### PR TITLE
ApplyPatchAction: Add option to deactivate creation of .orig files

### PR DESCRIPTION
**Pull request by abhsag24**
Fixes https://github.com/coala-analyzer/coala/issues/1186

### CODE REVIEW

**Comment by sils1297** [comment]
Hey, also I'd switch your commits, you'll first want to introduce the functionality and then add the user option. coala's configuration system allows the user to set *any* option even via command line arguments (see `-S` argument) so first introduce a functionality, then the CLI parameter for it.

File: coalib/results/result_actions/ApplyPatchAction.py [hunk 1]
```diff
@@ -21,7 +22,11 @@ def is_applicable(result, original_file_dict, file_diff_dict):
...
 
-    def apply(self, result, original_file_dict, file_diff_dict):
+    def apply(self,
+              result,
+              original_file_dict,
+              file_diff_dict,
+              no_orig: bool=False):
         """
         Apply the patch automatically.
         """
```
Comment (abhsag24) [inline coalib/results/result_actions/ApplyPatchAction.py]: unack didn't add documentation

**Comment by sils1297** [comment]
ack 15331df9f713e0205d64ed6cd556dcbe55d43071, 30cbd8a0e87, 937e187a3d4

**Comment by sils1297** [comment]
@rultor merge

**Pull request merged**